kafka_consumer/datadog_checks/kafka_consumer/client/confluent_kafka_client.py

```diff
@@ -74,7 +76,7 @@ def get_consumer_offsets(self):
         # {(consumer_group, topic, partition): offset}
         consumer_offsets = {}
 
-        if self.config._monitor_unlisted_consumer_groups:
+        if self.config._monitor_unlisted_consumer_groups or self.config._consumer_groups_regex:
```

Comment:
Moved to it's own method called `_get_consumer_groups()`, although I'll keep it as `return` for clarity since we are already yielding the output of `_get_consumer_offset_futures()`, which takes the output of `_get_consumer_groups()` as its own input.